### PR TITLE
[plugin.audio.navahang] v2.3

### PR DESCRIPTION
**Pull request by amirrammari**
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

**Comment by Rechi** [comment]
I don't think it's not Python 3 compatible, I know it.
I tested it with an Kodi version built against Python 3. Nightly builds are currently built with Python 2.7.

**Comment by amirrammari** [comment]
Hello @Rechi 
Now I think this add-on is compatible with Python 3.

File: plugin.audio.navahang/main.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+from xbmcplugin import addDirectoryItem, endOfDirectory
+
+ADDON = xbmcaddon.Addon()
+ADDONNAME = ADDON.getAddonInfo('name')
+ADDONID = ADDON.getAddonInfo('id')
+ADDONVERSION = ADDON.getAddonInfo('version')
+CWD = ADDON.getAddonInfo('path').decode("utf-8")
+LANGUAGE = ADDON.getLocalizedString
+
+DataLink = 'http://navahang.me/webservice/GetMp3?type=Featured&from=0&to=20'
```
Comment (amirrammari) [inline plugin.audio.navahang/main.py]:
```suggestion
DataLink = 'http://navahang.me/webservice/GetMp3?type=Featured&from=0&to=50'
```

File: plugin.audio.navahang/addon.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	<extension point="xbmc.addon.metadata">
+	<summary lang="en_GB">Plugin for Navahang</summary>
+	<description lang="en_GB">Navahang is a media company with one of the largest online archives, dedicated to Persian music along with 24/7 Live Radio.</description>
+	<disclaimer lang="en_GB">This plugin is not endorsed by Navahang LLC</disclaimer>
+	<summary lang="fa_IR">افزونه برای نواهنگ</summary>
+	<description lang="fa_IR">نواهنگ یک شرکت رسانه ای با یکی از بزرگترین آرشیو های آنلاین است که به موسیقی فارسی و همراه با ۲۴/۷ رادیو زنده اختصاص یافته است.</description>
+	<disclaimer lang="fa_IR">این افزونه توسط شرکت نواهنگ تایید نشده است</disclaimer>
+	<platform>all</platform>
+	<language>fa</language>
+	<license>GNU GENERAL PUBLIC LICENSE. Version 2, June 1991</license>
```
Comment (enen92) [inline plugin.audio.navahang/addon.xml]: Please use the spdx license identifier alternative. See here: https://spdx.org/licenses/

File: plugin.audio.navahang/main.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+DataLink = 'http://navahang.me/webservice/GetMp3?type=Featured&from=0&to=50'
+
+class LoadLister:
+
+        def __init__(self, url):
+                self.url = url
+
+        def addLink(self, name, url, image = '', info = {}, totalItems = 0):
+                item = ListItem(name.encode('utf-8'), iconImage = image, thumbnailImage = image)
```
Comment (enen92) [inline plugin.audio.navahang/main.py]: iconImage and thumbnailImage kwargs are deprecated. Please use only `listitem.setArt`

Comment (enen92) [inline plugin.audio.navahang/main.py]: `encode('utf-8´)` should not be needed under python3

File: plugin.audio.navahang/addon.xml (hunk 2 above)
Comment (amirrammari) [reply plugin.audio.navahang/addon.xml]:
```suggestion
	<license>GPL-2.0-only</license>
```

File: plugin.audio.navahang/main.py (hunk 3 above)
Comment (amirrammari) [reply plugin.audio.navahang/main.py]:
```suggestion
                item = ListItem(name, iconImage = image, thumbnailImage = image)
```

Comment (amirrammari) [reply plugin.audio.navahang/main.py]:
```suggestion
                item = ListItem(name).setArt({ 'poster': image, 'banner': image})
```

Comment (amirrammari) [reply plugin.audio.navahang/main.py]:
```suggestion
                item = ListItem(name).setArt({ 'poster': image, 'banner': image})
```

File: plugin.audio.navahang/main.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+DataLink = 'http://navahang.me/webservice/GetMp3?type=Featured&from=0&to=50'
+
+class LoadLister:
+
+        def __init__(self, url):
+                self.url = url
+
+        def addLink(self, name, url, image = '', info = {}, totalItems = 0):
+                item = ListItem(name, iconImage = image, thumbnailImage = image)
```
Comment (amirrammari) [inline plugin.audio.navahang/main.py]:
```suggestion
                item = ListItem(name, iconImage = image, thumbnailImage = image)
                item = ListItem(name).setArt({ 'poster': image, 'banner': image})
```

**Comment by enen92** [comment]
Please squash the commits for merging

**Comment by amirrammari** [comment]
> Please squash the commits for merging

How I can squash the commits? Please Help me.

**Comment by enen92** [comment]
Find the commit hash before your first commit using `git log`.
Then:
`git rebase -i HASH`
Keep the first line as pick and change all the other lines to `squash`. Save the file.
Then force push:
`git push -u origin leia --force`

**Comment by Rechi** [comment]
To fix your pull request execute the following commands (origin is your repo, upstream is the xbmc repo)

```
git fetch upstream
git checkout leia
git reset --hard upstream/leia
git checkout origin/leia -- plugin.audio.navahang/
git commit -m '[plugin.audio.navahang] v2.3'
git push -f origin leia
```

**Comment by amirrammari** [comment]
Hello @Rechi 
Travis tests have failed
Why? It's worked fine!

**Comment by Rechi** [comment]
That was a random failure which sadly happens from time to time.

I tested the add-on again and it's still not Python 3 compatible.

**Comment by amirrammari** [comment]
Hi @Rechi
Please test the add-on again.
I think now it's Python 3 compatible.
Thanks.

**Pull request merged**